packages/integration-aws-node/README.md

```diff
@@ -0,0 +1,105 @@
+# @backstage/integration-aws-node
+
+This package providers helpers for fetching AWS account credentials
+to be used by AWS SDK clients in backend packages and plugins.
+
+## Integrate new plugins
+
+Backend plugins can provide an AWS ARN or account ID to this library in order to
+retrieve a credentials provider for the relevant account that can be fed directly
+to an AWS SDK client.
+AWS SDK for Javascript V3 must be used.
+
+```typescript
+const awsCredentialsProvider = DefaultAwsCredentialsProvider.fromConfig(config);
+
+// provide the account ID explicitly
+const creds = await awsCredentialsProvider.getCredentials({ accountId });
+// OR extract the account ID from the ARN
+const creds = await awsCredentialsProvider.getCredentials({ arn });
+// OR provide neither to get main account's credentials
+const creds = await awsCredentialsProvider.getCredentials({});
+
+// Example constructing an AWS Proton client with the returned credentials provider
+const client = new ProtonClient({
+  region,
+  credentialDefaultProvider: () => creds.provider,
+});
+```
+
+## Backstage app configuration
+
+Users of plugins and packages that use this library
+will configure their AWS account information and credentials in their
+Backstage app config.
+Users can configure IAM user credentials, IAM roles, and profile names
+for their AWS accounts in their Backstage config.
+
+If the AWS integration configuration is missing, the credentials provider
+from this package will fall back to the AWS SDK default credentials chain for
+resources in the main AWS account.
+The default credentials chain for Node resolves credentials in the
+following order of precedence:
+
+1. Environment variables
+2. SSO credentials from token cache
+3. Web identity token credentials
+4. Shared credentials files
+5. The EC2/ECS Instance Metadata Service
+
+See more about the AWS SDK default credentials chain in the
+[AWS SDK for Javascript Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide/setting-credentials-node.html).
+
+Configuration examples:
+
+```yaml
+integrations:
+  aws:
+    # The main account is used for assume-role calls into other AWS accounts.
+    # It can be omitted to fall back to the AWS SDK's default creds chain.
+    mainAccount:
+      accessKeyId: ${MY_ACCESS_KEY_ID}
+      secretAccessKey: ${MY_SECRET_ACCESS_KEY}
+
+    # Account credentials can be configured individually per account
+    accounts:
```

Comment:
personally, I found array always a bit inconvenient regarding the use of env vars to set or overwrite values. Alternatively, we could use e.g. the accountId as key. Just a thought, no blocker.

Reply:
I could go either way, but this seemed to be the convention that other integrations followed.  Account IDs do also get weird as yaml keys, since they look like integers but are actually strings and always have to be quoted in yaml.

```yaml
  accounts:
    '012345678999':
        roleName: 'my-iam-role-name'
```
vs
```yaml
  accounts:
    - accountId: '012345678999' # Still has to be quoted, but seems less icky than quoting a key
       roleName: 'my-iam-role-name'
```